bumble/gatt_server.py

```diff
@@ -66,6 +66,15 @@ def send_gatt_pdu(self, connection_handle, pdu):
     def next_handle(self):
         return 1 + len(self.attributes)
 
+    def get_advertising_service_data(self):
+        service_advertising_data_map = dict[Service, bytes]()
+        for attribute in self.attributes:
+            if isinstance(attribute, Service):
+                if not attribute.get_advertising_data():
```

Comment:
this is sub-optimal, as it will call `get_advertising_data()` twice when not None.
Consider instead using the 'walrus' operator and using a comprehension to construct the dictionary, with something like:
```
return {
    attribute: data for attribute in self.attributes 
    if isinstance(attribute, Service) and (data := attribute.get_advertising_data())
}

Reply:
ack